### PR TITLE
Intern Task Alina Mulik: Homework

### PR DESCRIPTION
**Pull request by alina-mulik**
added homework #13 files

### CODE REVIEW

File: Homework11/Common/Drivers/WebDriverFactory.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using Homework11.Data;
+using OpenQA.Selenium;
+using System.Collections.Concurrent;
```
Comment (yauhenists) [inline Homework11/Common/Drivers/WebDriverFactory.cs:3]: Ordering for usings:
1. All system
2. All other rest
Each group should be sorted alphabetically

Please check other files

File: Homework11/Common/Drivers/WebDriverFactory.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        // create an instance of Actions
+        public static Actions Actions => new Actions(Driver);
+
+        // create an instance of IJavaScriptExecutor
+        public static IJavaScriptExecutor JavaScriptExecutor => (IJavaScriptExecutor)Driver;
+
+        public static void QuitDriver() => Driver.Quit();
+
+        public static string GetCurrentUrl() => Driver.Url;
```
Comment (yauhenists) [inline Homework11/Common/Drivers/WebDriverFactory.cs:41]: This method is never used

File: Homework11/Common/Drivers/WebDriverFactory.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public class WebDriverFactory
+    {
+        // collection to isolate threads for possible parallelization
+        private static readonly ConcurrentDictionary<string, IWebDriver> DriverCollection = new ConcurrentDictionary<string, IWebDriver>();
+
+        public static IWebDriver Driver
+        {
+            get
+            {
+                // if driver is not yet initialized, we initalize it
```
Comment (yauhenists) [inline Homework11/Common/Drivers/WebDriverFactory.cs:19]: Typo in comment

File: Homework11/Common/Extensions/WebDriverExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿using Homework11.Common.Drivers;
```
Comment (yauhenists) [inline Homework11/Common/Extensions/WebDriverExtensions.cs:1]: Please remove unnecessary using and check other files

File: Homework11/Common/Extensions/WebDriverExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+
+namespace Homework11.Common.Extensions
+{
+    public static class WebDriverExtensions
+    {
+        // extension method to get webdriver wait
+        // "this IWebDriver driver" shows that extends IWebDriver method extends IWebDriver functionality
+        public static WebDriverWait GetWebDriverWait(this IWebDriver driver, int timeoutseconds = 15, TimeSpan? pollingInterval = null, params Type[] exceptionTypes)
```
Comment (yauhenists) [inline Homework11/Common/Extensions/WebDriverExtensions.cs:11]: Please follow the naming rules: timeoutseconds -> timeoutSeconds

File: Homework11/Common/WebElements/DemoQaWebElement.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+        public string GetDomProperty(string propertyName) => WebElement.GetDomProperty(propertyName);
+
+        public ISearchContext GetShadowRoot() => WebElement.GetShadowRoot();
+
+        public void SendKeys(string text) => WebElement.SendKeys(text);
+
+        public void Submit() => WebElement.Submit();
+
+        // method to scroll elemet into view using JavaScript
```
Comment (yauhenists) [inline Homework11/Common/WebElements/DemoQaWebElement.cs:81]: Typo in comment

**Review by yauhenists — CHANGES_REQUESTED** [review]

File: Homework11/Common/Drivers/WebDriverFactory.cs (hunk 1 above)
Comment (alina-mulik) [reply Homework11/Common/Drivers/WebDriverFactory.cs:3]: fixed

File: Homework11/Common/Drivers/WebDriverFactory.cs (hunk 2 above)
Comment (alina-mulik) [reply Homework11/Common/Drivers/WebDriverFactory.cs:41]: fixed

File: Homework11/Common/Drivers/WebDriverFactory.cs (hunk 3 above)
Comment (alina-mulik) [reply Homework11/Common/Drivers/WebDriverFactory.cs:19]: fixed

File: Homework11/Common/Extensions/WebDriverExtensions.cs (hunk 4 above)
Comment (alina-mulik) [reply Homework11/Common/Extensions/WebDriverExtensions.cs:1]: fixed

File: Homework11/Common/Extensions/WebDriverExtensions.cs (hunk 5 above)
Comment (alina-mulik) [reply Homework11/Common/Extensions/WebDriverExtensions.cs:11]: fixed

File: Homework11/Common/WebElements/DemoQaWebElement.cs (hunk 6 above)
Comment (alina-mulik) [reply Homework11/Common/WebElements/DemoQaWebElement.cs:81]: fixed

**Pull request closed without merging**